### PR TITLE
Switch to platform constants from openshfit/api 

### PR DESCRIPTION
**Pull request by soltysh**
Hmm... I'm not quite sure this will work, but I'll need to give it a try.

/assign @deads2k

### CODE REVIEW

File: pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go [hunk 1]
```diff
@@ -73,19 +74,19 @@ func ObserveCloudProviderNames(genericListers configobserver.Listers, recorder e
 		glog.Warning("configmap/cluster-config-v1.kube-system: install-config.platform not found")
```
Comment (deads2k) [inline pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go]: You should not be reading this from a configmap anymore

File: pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go [hunk 2]
```diff
@@ -27,71 +22,53 @@ func ObserveCloudProviderNames(genericListers configobserver.Listers, recorder e
...
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(currentCloudProvider) > 0 {
+		if err := unstructured.SetNestedStringSlice(prevObservedConfig, currentCloudProvider, cloudProvidersPath...); err != nil {
+			return prevObservedConfig, append(errs, err)
 		}
 	}
 
+	if !listers.InfrastructureSynced() {
```
Comment (deads2k) [inline pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go]: remove.  We no longer need this because of cluster-config-operator

**Comment by deads2k** [comment]
just the comments about syncing.  lgtm otherwise.

File: pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go (hunk 2 above)
Comment (sanchezl) [reply pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go]: @soltysh https://github.com/soltysh/cluster-kube-controller-manager-operator/pull/1

File: pkg/operator/configobservation/interfaces.go [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
 )
 
 type Listers struct {
-	ConfigmapLister corelistersv1.ConfigMapLister
+	ConfigmapLister      corelistersv1.ConfigMapLister
+	InfrastructureLister configlistersv1.InfrastructureLister
+
+	InfrastructureSynced cache.InformerSynced
```
[the remark this answers is not in the document]
Comment (sanchezl) [reply pkg/operator/configobservation/interfaces.go]: @soltysh https://github.com/soltysh/cluster-kube-controller-manager-operator/pull/1

File: pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go (hunk 2 above)
Comment (soltysh) [reply pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go]: Thanks Luis :heart:

**Comment by soltysh** [comment]
@sanchezl ptal

**Comment by deads2k** [comment]
got fixed at some point

**Pull request closed without merging**